src/error-snippets/utils.ts

```diff
@@ -0,0 +1,52 @@
+import chalk from "chalk";
+import path from "path";
+import { fileURLToPath } from "url";
+import fs from "fs-extra";
+import { SOURCE_MAP_URL_COMMENT } from "./constants";
+
+interface formatFileNameHeaderOpts {
+    line: number;
+    linesAbove: number;
+    linesBelow: number;
+}
+
+export const softFileURLToPath = (fileName: string): string => {
+    if (!fileName.startsWith("file://")) {
+        return fileName;
+    }
+
+    try {
+        return fileURLToPath(fileName);
+    } catch (_) {
+        return fileName;
+    }
+};
+
+export const formatFileNameHeader = (fileName: string, opts: formatFileNameHeaderOpts): string => {
+    const lineNumberWidth = String(opts.line - opts.linesAbove).length;
+    const offsetWidth = String(opts.line + opts.linesBelow).length;
+
+    const lineNumberOffset = ".".repeat(lineNumberWidth).padStart(offsetWidth);
+    const offset = `  ${lineNumberOffset} |`;
+
+    return chalk.gray(offset + ` // ${fileName}\n` + offset + "\n");
+};
+
+export const getSourceCodeFile = async (fileName: string): Promise<string> => {
+    fileName = softFileURLToPath(fileName);
+
+    if (path.isAbsolute(fileName)) {
+        return fs.readFile(fileName, "utf8");
+    }
+
+    const response = await fetch(fileName);
+    const responseText = await response.text();
+
+    if (responseText.includes(SOURCE_MAP_URL_COMMENT) || !response.headers.has("SourceMap")) {
+        return responseText;
+    }
+
+    const sourceMapUrl = response.headers.get("SourceMap");
+
+    return responseText + "\n" + SOURCE_MAP_URL_COMMENT + sourceMapUrl;
```

Comment:
By specification, source maps could be in HTTP Header `SourceMap`.

If it is, we could just append our file contents and dont think about it elsewhere